packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -11,10 +11,824 @@ import 'picker.dart';
 /// Default aesthetic values obtained by comparing with iOS pickers.
 const double _kItemExtent = 32.0;
 const double _kPickerWidth = 330.0;
+const bool _kUseMagnifier = true;
+const double _kMagnification = 1.1;
+const double _kDatePickerPadSize = 12.0;
 /// Considers setting the default background color from the theme, in the future.
 const Color _kBackgroundColor = CupertinoColors.white;
 
 
+// Lays out the date picker based on how much space each single column needs.
+//
+// Each column is a child of this delegate, indexed from 0 to number of columns - 1.
+// Each column will be padded horizontally by 12.0 both left and right.
+//
+// The picker will be placed in the center, and the leftmost and rightmost
+// column will be extended equally to the remaining width.
+class _DatePickerLayoutDelegate extends MultiChildLayoutDelegate {
+  _DatePickerLayoutDelegate({
+    @required this.columnWidths,
+    @required this.textDirectionFactor,
+  }) : assert(columnWidths != null),
+       assert(textDirectionFactor != null);
+
+  // The list containing widths of all columns.
+  final List<double> columnWidths;
+
+  // textDirectionFactor is 1 if text is written left to right, and -1 if right to left.
+  final int textDirectionFactor;
+
+  @override
+  void performLayout(Size size) {
+    double remainingWidth = size.width;
```

Comment:
Reading this logic, since you already know how wide each column should be, is this really different from just having a Row where the first and last items have an expanded with flex 1 and everything else has a sizedbox with a padding?

i.e. do we need a CustomMultiChildLayout?